modules/kernel/src/main/scala-2.10/notebook/ReplCalculator.scala

```diff
@@ -222,14 +222,12 @@ class ReplCalculator(
         // this is safe in recent spark versions, see SPARK-6414
         log.debug(s"Interrupting the cell: $killCellId")
         val jobGroupId = JobTracking.jobGroupId(killCellId)
-        try {
+        // make sure sparkContext is already available!
+        if (repl.interp.allDefinedNames.filter(_.toString == "sparkContext").nonEmpty) {
```

Comment:
and this don't look trivial, e.g. in `init.sc` just calling `globalScope.reset()` before the imports do not fix it.
```scala
globalScope.reset()
import globalScope.{sparkContext, reset, sc}
```
so maybe the easiest fix would be to only remove jobs from queue if they are still there (if user clicked to run the cell multiple times, it would not kill the running jobs, which is not so bad...).